my-dsl.md

```diff
@@ -2,37 +2,102 @@
 _What is the name of the language? Link the name to its webpage 
 (if appropriate)._
 
+[JsonDSL](https://github.com/lift/lift/tree/master/framework/lift-base/lift-json) 
+(part of the Liftweb framework for Scala)
 
 # Domain
 _Describe the language's domain in five words._
 
+Building name/value data structures
 
 # Computational model
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of 
 what happens when a program in your DSL runs._
 
+The code with the JSON's information goes through a parser that builds the JSON. 
+The program creates key/value pairs or concatenates when it sees parentheses or 
+~'s. Within each pair, implicit methods convert typical Scala types (i.e. Int, 
+String) into JSON types (i.e. JInt, JString). Finally, we get a JSON object 
+containing the name/value pairs.
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+This DSL is purely domain-specific because it has very limited capabilities. 
+It can only build JSONS by parsing through information, narrowing the domain 
+focus to strictly this data structure. Moreover, it lacks conditions, loops, 
+and variables, a common trait of DSLS according to Fowler (page 30).
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+I spent a lot of time thinking about this question because JsonDSL isn't 
+remarkably different from the Scala language, but at the same time, it's not 
+valid Scala code without the parsers provided in the library. Maybe it is 
+because Scala is so DSL-heavy. 
+
+Though the lines are blurry, I think JsonDSL is external because it relies on 
+parsing custom syntax. In addition, on page 30, Fowler states that "an internal 
```

Comment:
Haven't dug into the language so I don't know the details about how it does the parsing, but Scala does have super powerful abilities to write code that doesn't _look_ like it is valid Scala but _is_ valid Scala.  As an example, [here is our internal Piconot DSL from last year](https://github.com/hmc-cs111-fall2015/piconot/pull/6/files)

```scala
If you hold weapon 1
and Shire has anything
and LonelyMountain has anything
and UndyingLands is empty
and Mordor has anything
then go towards UndyingLands and ready weapon 1,
       
If you hold weapon 1
and Shire has anything
and LonelyMountain has anything
and UndyingLands has Orcs
and Mordor has anything
then stay and ready weapon 2,
```

That continues for a while... but hopefully it gives you an idea of how powerful Scala is.